apps/ejabberd/src/mod_caps.erl

```diff
@@ -227,6 +227,11 @@ disco_info(Acc, Host, Module, Node, Lang) ->
 
 c2s_presence_in(C2SState,
                 {From, To, {_, _, Attrs, Els}}) ->
+    % XXX it is never called, it appears, the only call has a different
+    % signature:
+    % ejabberd_hooks:run_fold(c2s_presence_in, StateData#state.server,
+    %    StateData, [{From, To, Packet}]),
+    % where Packet is #xmlel
```

Comment:
Right packet is #xml so in the vm is actually a tuple:
```erlang
{xmlel, Name, Attrs, Els}
```
and matches this patter.

Also according to our coverage stats here: https://coveralls.io/builds/10600399/source?filename=apps%2Fejabberd%2Fsrc%2Fmod_caps.erl#L228 you see that this part of the code is called.